analyzers/tests/SonarAnalyzer.Test/Helpers/DisjointSetsPrimitivesTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿using static SonarAnalyzer.Helpers.DisjointSetsPrimitives;
+
+namespace SonarAnalyzer.Test.Helpers;
+
+[TestClass]
+public class DisjointSetsPrimitivesTest
+{
+    [TestMethod]
+    public void FindRootAndUnion_AreConsistent()
+    {
+        var parents = Enumerable.Range(1, 5).ToDictionary(i => i.ToString(), i => i.ToString());
+        foreach (var item in parents.Keys)
+        {
+            FindRoot(parents, item).Should().Be(item);
+        }
+
+        Union(parents, "1", "1");
+        FindRoot(parents, "1").Should().Be("1");                    // Reflexivity
+        Union(parents, "1", "2");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Correctness
+        Union(parents, "1", "2");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Idempotency
+        Union(parents, "2", "1");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "2")); // Symmetry
+
+        FindRoot(parents, "3").Should().Be("3");
+        Union(parents, "2", "3");
+        FindRoot(parents, "2").Should().Be(FindRoot(parents, "3"));
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "3")); // Transitivity
+        Union(parents, "3", "4");
+        FindRoot(parents, "1").Should().Be(FindRoot(parents, "4")); // Double transitivity
+        Union(parents, "4", "1");
+        FindRoot(parents, "4").Should().Be(FindRoot(parents, "1")); // Idempotency after transitivity
```

Comment:
Similar to the other comment, optional as well:

```csharp
    private static void RootsShouldBeEqual(Dictionary<string, string> parents, string left, string right) =>
        FindRoot(parents, left).Should().Be(FindRoot(parents, right));
```

...and use it like so:

```csharp
RootsShouldBeEqual(parents, "4", "1"); // Idempotency after transitivity
```

Reply:
In this case, I don't see the point, and actually, I think it makes the code a bit more obscure. 
`FindRoot` is not clutter, it is the method under test. 
Hiding away the thing being tested in a utility method to have few characters less, seems to have a negative return. 

In the case above we could spare for many cases the `).Should().BeEquivalentTo<List<string>>(`, that was just unnecessary clutter. Hence the utility of `DisjointSetShouldBe`. In this case, however, you are basically sparing `).Should().Be(`.

That is why I wouldn't make such a change.